ledger-core/virtual/integration/utils/vcallrequest.go

```diff
@@ -67,6 +67,12 @@ func (h *VCallRequestConstructorHandler) SetClass(ref reference.Global) {
 	h.regenerate()
 }
 
+//nolint:interfacer
+func (h *VCallRequestConstructorHandler) SetCaller(ref reference.Global) {
+	h.request.Caller.Set(ref)
+	h.regenerate()
```

Comment:
well, a Builder pattern fits better here, instead of regenerating on each Set

Reply:
I suggest to do it later if we'll need it in the future